plugins/kosync.koplugin/main.lua

```diff
@@ -277,31 +304,47 @@ function KOSync:login()
                     enabled = true,
                     callback = function()
                         local username, password = self:getCredential()
-                        self:closeDialog()
-                        UIManager:scheduleIn(0.5, function()
-                            self:doLogin(username, password)
-                        end)
-
-                        UIManager:show(InfoMessage:new{
-                            text = _("Logging in. Please wait…"),
-                            timeout = 1,
-                        })
+                        local ok, err = validateUser(username, password)
+                        if not ok then
+                            UIManager:show(InfoMessage:new{
+                                text = err,
+                                timeout = 3,
+                            })
+                        else
+                            self:closeDialog()
+                            UIManager:scheduleIn(0.5, function()
```

Comment:
Seems like a very long delay. Is this still in line with similar code?

Reply:
2s is enough. 1s could be but not if the user doesn't expect a message here.